### ISSUE TITLE
replace frozendict with immutabledict

### ISSUE THREAD

**Issue by VanNostrand** [ranaroussi/yfinance#1752]
### Describe bug

Hi, Gentoo Linux is going to remove the python package frozendict:
https://bugs.gentoo.org/918899
They write "Originally added for net-im/synapse, which switched to dev-python/immutabledict since.  Hacky C extension that supports up to Python 3.10". "Removal on 2023-12-30."

The project site of immutabledict says it is a fork of frozendict, written in pure python: https://github.com/corenting/immutabledict/
The package did not have additional dependencies as I installed it.

I tried the following change in yfinance/data.py and yfinance seemed to work (the script I wrote with it did everything I expected from it):
````python
#from frozendict import frozendict
from immutabledict import immutabledict
````

### Simple code that reproduces your problem

see bug description

### Debug log

There are no errors:
DEBUG    get_raw_json(): https://query2.finance.yahoo.com/v10/finance/quoteSummary/AMZ.DE
DEBUG    Entering get()
DEBUG     url=https://query2.finance.yahoo.com/v10/finance/quoteSummary/AMZ.DE
DEBUG     params={'modules': 'financialData,quoteType,defaultKeyStatistics,assetProfile,summaryDetail', 'ssl': 'true'}
DEBUG     Entering _get_cookie_and_crumb()
DEBUG      cookie_mode = 'basic'
DEBUG      Entering _get_cookie_and_crumb_basic()
DEBUG       loaded persistent cookie
DEBUG       reusing cookie
DEBUG       crumb = 'SnU9.2cbXIN'
DEBUG      Exiting _get_cookie_and_crumb_basic()
DEBUG     Exiting _get_cookie_and_crumb()
DEBUG    Exiting get()

### `yfinance` version

yfinance-0.2.32

### Python version

Python 3.11.6

### Operating system

Linux

**Comment by VanNostrand** [comment]
> #1084

well, I could fork, change that one line and create a PR but I am not a python developer and pretty clueless about that "fix" and about how yfinance works.
For example, the immutabledict package should have a function name "frozendict" to really work as a drop-in replacement - but it hasn't. So I wonder if my "fix" only made python shut up in regards of the missing package and I was just lucky that no frozendict related code was executed.
A quick grep has shown me, that the word frozendict also accurs in some meta/requirement/setup files. I have even less clue about how that works.
So whatever my quick hack does, it will probably break a lot.

**Comment by VanNostrand** [comment]
Yeah, I just completely commented out the import line and it still works... Sorry, but I am not going to touch the code officially, I don't know what I am doing.

**Comment by dokson** [comment]
@VanNostrand opened #2821 to drop the `frozendict` dependency entirely. The single use case (making dict args hashable for `lru_cache`) is covered by a ~15-line internal `FrozenDict` in `yfinance/utils.py`, so no replacement package is needed — yfinance becomes installable on Gentoo (and anywhere else `frozendict` is unavailable) out of the box.
